token/js/src/instructions/uiAmountToAmount.ts

```diff
@@ -0,0 +1,123 @@
+import { u8, struct, cstr} from '@solana/buffer-layout';
+import { AccountMeta, PublicKey, TransactionInstruction } from '@solana/web3.js';
+import { TOKEN_PROGRAM_ID } from '../constants';
+import {
+    TokenInvalidInstructionDataError,
+    TokenInvalidInstructionKeysError,
+    TokenInvalidInstructionProgramError,
+    TokenInvalidInstructionTypeError,
+} from '../errors';
+import { TokenInstruction } from './types';
+
+/** TODO: docs */
+export interface UiAmountToAmountInstructionData {
+    instruction: TokenInstruction.UiAmountToAmount;
+    amount: string;
+}
+
+/** TODO: docs */
+export const uiAmountToAmountInstructionData = struct<UiAmountToAmountInstructionData>([u8('instruction'),cstr('amount')]);
```

Comment:
When taking ```cstr``` for amount i am getting error ``` RangeError [ERR_INVALID_ARG_VALUE]: The argument 'size' is invalid. Received -1```, what should i use for representing as ```uiamount``` is string

Reply:
I think that `utf8` will be a better option, since `cstr` will null-terminate the bytes, and the program isn't expecting that

Reply:
So i found that the Error is due to ```Buffer.alloc(uiAmountToAmountInstructionData.span)``` the span generated here is -1 so it is throwing error. i tried using utf8 but still i am getting -1 as span . I tried something like 
```
export interface UiAmountToAmountInstructionData {
    instruction: TokenInstruction.UiAmountToAmount;
    amount: string | LayoutObject;
}

export const uiAmountToAmountInstructionData = struct<UiAmountToAmountInstructionData([u8('instruction'),utf8(128,'amount')]);
```

Reply:
Ah right my apologies, yes, you'll need to specify the size somehow, otherwise `span` will give you that `-1` result.  Instead of doing `Buffer.alloc(uiAmountToAmountInstructionData.span);`, you can do:

```
const buf = Buffer.from(amount, 'utf8');
const uiAmountToAmountInstructionData = struct<UiAmountToAmountInstructionData([u8('instruction'), blob(buf.length)]);
const data = Buffer.alloc(uiAmountToAmountInstructionData.span);
```

And then the rest as normal